spirv_glsl.cpp

```diff
@@ -14865,11 +14865,19 @@ void CompilerGLSL::emit_block_chain(SPIRBlock &block)
 
 		size_t num_blocks = block_declaration_order.size();
 
-		const auto to_case_label = [](uint32_t literal, bool is_unsigned_case) -> string {
-			return is_unsigned_case ? convert_to_string(literal) : convert_to_string(int32_t(literal));
+		// Precompute the is_32bit_int since it's block based, not per-case
+		// based so we then move the bool to the lambda.
+		const bool is_32bit_int = (type.basetype == SPIRType::Int);
+		const auto to_case_label = [=](uint64_t literal, bool is_unsigned_case) -> string {
```

Comment:
Could just be `uint64_t literal, uint32_t width, bool is_unsigned_case` and then just roll out the 4 cases.